src/backend/kvm/thread.rs

```diff
@@ -77,77 +77,132 @@ impl<P: KeepPersonality> Thread<P> {
             .as_virt()
             .start;
 
-        Ok([vaddr.as_u64().into(), 0.into()])
+        Ok(vaddr.as_u64() as _)
     }
 
-    pub fn meminfo(&self, block: &mut Block) -> Result<[Register<usize>; 2], i32> {
+    pub fn meminfo(&self) -> sallyport::Result<usize> {
         let keep = self.keep.read().unwrap();
 
         // The maximum number of memory slots possible for a virtual machine
         // minus the ones which were already used.
-        let mem_slots = keep.kvm_fd.get_nr_memslots() - keep.regions.len();
-
-        // FIXME:
-        // Obsolete, if [host side syscall verification and address translation](https://github.com/enarx/enarx/issues/957)
-        // is implemented.
-        let virt_start = Address::from(keep.sallyport_start.as_mut_ptr());
+        Ok(keep.kvm_fd.get_nr_memslots() - keep.regions.len())
+    }
 
-        let mem_info: MemInfo = MemInfo {
-            virt_start,
-            mem_slots,
-        };
+    fn kvm_enarxcall<'a>(
+        &mut self,
+        enarxcall: &'a mut Payload,
+        data: &'a mut [u8],
+    ) -> Result<Option<Item<'a>>> {
+        match enarxcall {
+            item::Enarxcall {
+                num: item::enarxcall::Number::MemInfo,
+                ret,
+                ..
+            } => {
+                *ret = match self.meminfo() {
+                    Ok(n) => n as usize,
+                    Err(e) => -e as usize,
+                };
+                Ok(None)
+            }
 
-        let c = block.cursor();
-        c.write(&mem_info).map_err(|_| libc::ENOBUFS)?;
+            item::Enarxcall {
+                num: item::enarxcall::Number::BalloonMemory,
+                argv: [log2, npgs, addr, ..],
+                ret,
+            } => {
+                *ret = match self.balloon(*log2, *npgs, *addr) {
+                    Ok(n) => n as usize,
+                    Err(e) => -e as usize,
+                };
+                Ok(None)
+            }
 
-        Ok([0.into(), 0.into()])
+            _ => return Ok(Some(Item::Enarxcall(enarxcall, data))),
+        }
     }
 }
 
 impl<P: KeepPersonality> super::super::Thread for Thread<P> {
-    fn enter(&mut self) -> Result<Command<'_>> {
+    fn enter(&mut self, _gdblisten: &Option<String>) -> Result<Command> {
```

Comment:
drop the `_`

Reply:
because of unused arguments, if feature gdb is not turned on